### PR TITLE
Stealth materials

### PR DESCRIPTION
**Pull request by SuicidalInsanity**
-Radar stealth coatings - can now apply radar absorbent/reflective materials on a per part basis.
	-Armor coatings less than 10mm thick will provide lessened benefits.
-Adds new radarReflectivity field to Armor materials so specify degree of radar absorption/reflectivity of the armor.
-Adds new radarMod field to Hull materials to set degree of radar return off unarmored parts.
-Adds new Radar Absorbent Coating armor type
-Adds new Radar Absorbent Foam hull material.

### CODE REVIEW

File: BDArmory/Damage/HitpointTracker.cs [hunk 1]
```diff
@@ -1557,6 +1568,31 @@ void SetHullMass()
             }
             _hullConfigured = true;
         }
+        private void CalculateRCSreduction()
+        {
+            float radarReflected = armorRadarReturnFactor < 10 ? 1 - (Mathf.Log(Mathf.Max(ArmorThickness, 1f), 10) * (1 - armorRadarReturnFactor)) : //armor < 10 will have reduced radar absorbsion
```
Comment (BrettRyland) [inline BDArmory/Damage/HitpointTracker.cs:1573]: Using `Mathf.Log10(x)` is faster than `Mathf.Log(x, 10)` (0.00818µs vs 0.0132µs) due to `Log(10)` being a predefined constant.

File: BDArmory/Radar/RadarUtils.cs [hunk 2]
```diff
@@ -716,7 +717,54 @@ public static TargetInfo RenderVesselRadarSnapshot(Vessel v, Transform t, Target
                     v.SetPosition(v.transform.position - presentationPosition);
                 }
             }
-
+            if (!BDArmorySettings.DEBUG_RADAR)
```
Comment (BrettRyland) [inline BDArmory/Radar/RadarUtils.cs:720]: Is it intentional that this part gets called when you're _not_ debugging radar?
I suspect it is, but thought it best to ask.

**Review by BrettRyland — COMMENTED** [review]
I don't see any issues with the PR currently. Are you still working on it or is it ready to merge?

**Comment by SuicidalInsanity** [comment]
It should be ready to merge

**Pull request merged**